core/controllers/topic_editor.py

```diff
@@ -149,8 +151,45 @@ def get(self, topic_id: str) -> None:
         additional_story_summary_dicts = [
             summary.to_dict() for summary in additional_story_summaries]
 
+        canonical_stories_ids = [summary['id'] for
+            summary in canonical_story_summary_dicts]
+        canonical_stories = story_fetchers.get_stories_by_ids(
+            canonical_stories_ids)
+        non_empty_canonical_stories = [story for story in
+            canonical_stories if story is not None]
         updated_canonical_story_summary_dicts = []
         for summary in canonical_story_summary_dicts:
+            story = None
+            for canonical_story in non_empty_canonical_stories:
+                if canonical_story.id == summary['id']:
+                    story = canonical_story
+            if story is None:
+                continue
+            nodes = story.story_contents.nodes
+            total_chapters_count = len(nodes)
+            published_chapters_count = 0
+            upcoming_chapters_count = 0
+            overdue_chapters_count = 0
+            upcoming_chapters_expected_days = []
+            for node in nodes:
+                if node.status == constants.STORYNODE_STATUS_PUBLISHED:
+                    published_chapters_count += 1
+                if (node.status != constants.STORYNODE_STATUS_PUBLISHED and
+                    node.planned_publication_date is not None):
+                    current_time = (
+                        utils.convert_millisecs_time_to_datetime_object(
+                        utils.get_current_time_in_millisecs() -
+                        1000.0 * time.timezone))
+                    if (current_time < node.planned_publication_date
+                        < current_time + datetime.timedelta(
+                        constants.UPCOMING_CHAPTERS_DAY_LIMIT)):
```

Comment:
What hapens if these values are equal to the boundary conditions?

Reply:
The conditions will simply be false. Actually the logic behind this code has been verified in my GSoC proposal.

Reply:
Can you refer me to that part?